### PR TITLE
Allow initializing Zaplib without a canvas or with your own canvas

### LINKED ISSUE #20: Using Zaplib without a canvas
Currently we always create a canvas when using Zaplib. That's a bit annoying if you just want to do some computation. We can probably make the canvas part of `defaultStyles`, and if that is set to false allow the user to specify it (or omit it).

### PR DESCRIPTION
**Pull request by janpaul123**
This is helpful for when using Zaplib for pure computation: #20 and #18.

It’s also a first step toward giving users more control over how
rendering works exactly (e.g. multiple canvases; different layering;
etc).

Tested with manually `tutorial_3d_rendering_step_2` which does pure
computation. Will be better covered in CI when we do #29.

Also added a bunch of docs.

Closes #20.

### CODE REVIEW

File: zaplib/web/types.ts [hunk 1]
```diff
@@ -1,5 +1,6 @@
 export type Initialize = (initParams: {
   filename: string;
+  canvas?: HTMLCanvasElement;
   baseUri?: string;
   defaultStyles?: boolean;
```
Comment (hhsaez) [inline zaplib/web/types.ts:5]: Is it worth adding another option for `defaultStyles` minus the logo?

Comment (janpaul123) [reply zaplib/web/types.ts:5]: Ah interesting, what are you thinking here? My thought was that `defaultStyles` is a handy shortcut for our examples, but that if you want to actually use Zaplib in production you probably want to have more control over things anyway?

Comment (janpaul123) [reply zaplib/web/types.ts:5]: Yeah we could do that, but I guess what I'm thinking is that in most cases you'd probably want more control over the canvas anyway, so I don't know how useful that would be?

Comment (janpaul123) [reply zaplib/web/types.ts:5]: It's also only two lines of code to do this yourself:

```js
const canvas = document.createElement("canvas");
document.body.appendChild(canvas);
zaplib.initialize({ canvas });
```

Maybe I should add this to the docs though to make it easy for people?

File: zaplib/web/wasm_runtime.ts [hunk 2]
```diff
@@ -311,10 +310,66 @@ export const initialize: Initialize = (initParams) => {
         addLoadingIndicator();
       }
 
+      // One of these variables should get set, depending on if
+      // the browser supports OffscreenCanvas or not. Or neither,
+      // if we're not rendering at all. But they should not both get set!
```
Comment (pankdm) [inline zaplib/web/wasm_runtime.ts:315]: Should we add an assertion somewhere in the end for this?

Comment (janpaul123) [reply zaplib/web/wasm_runtime.ts:315]: Yeah good idea 👍🏻

File: zaplib/web/wasm_runtime.ts [hunk 3]
```diff
@@ -424,220 +479,189 @@ export const initialize: Initialize = (initParams) => {
...
-        ) {
-          event.preventDefault();
-        }
-      });
+      let canvas: HTMLCanvasElement | undefined = initParams.canvas;
+      if (!canvas && initParams.defaultStyles) {
+        canvas = document.createElement("canvas");
+        document.body.appendChild(canvas);
+      }
+      if (canvas) {
```
Comment (pankdm) [inline zaplib/web/wasm_runtime.ts:503]: This `if (canvas) ` block is now ~160 lines of code, which makes it hard to follow where it ends / starts next code. Maybe we could extract this into a separate function?

Comment (janpaul123) [reply zaplib/web/wasm_runtime.ts:503]: Yeah, I like that! Will do.

**Review by pankdm — APPROVED** [review]
Looks very good! Just a few nits inline.
